File: typescript/tracing/azure-openai/01_chat_completion.ts
import {
  createAzureClient,
  createRespan,
  installMockAzureOpenAIResponses,
  logExampleResult,
  runWithExampleTrace,
} from "./_shared.js";

const workflowName = "TypeScript Azure OpenAI Chat Example";

export async function chatCompletionExample(): Promise<void> {
  const restoreMocks = installMockAzureOpenAIResponses();
  const respan = createRespan("typescript-azure-openai-chat-example");
  await respan.initialize();

  try {
    const result = await runWithExampleTrace(respan, workflowName, async () => {
      const client = createAzureClient("gpt-4o-mini");
      return await client.chat.completions.create({
        model: "gpt-4o-mini",
        messages: [
          { role: "system", content: "You are concise." },
          { role: "user", content: "Confirm Azure OpenAI tracing is active." },
        ],
        extraAttributes: {
          "respan.metadata.azure_feature": "chat",
        },
      } as any);
    });

    logExampleResult(workflowName, {
      content: result.choices[0]?.message?.content,
      model: result.model,
    });
  } finally {
    await respan.shutdown();
    restoreMocks();
  }
}

await chatCompletionExample();

File: python/tracing/groq/01_chat_completion.py
from __future__ import annotations

from _shared import (
    example_attributes,
    make_client,
    make_custom_identifier,
    make_respan,
    model_name,
    print_result,
    set_workflow_input,
    workflow_name,
)
from respan import workflow

EXAMPLE_NAME = "chat-completion"


@workflow(name=workflow_name(EXAMPLE_NAME))
def _chat_completion_workflow(client) -> str:
    prompt = "Reply with one concise sentence about Groq tracing."
    set_workflow_input(prompt)
    response = client.chat.completions.create(
        model=model_name(),
        messages=[{"role": "user", "content": prompt}],
        temperature=0,
    )
    return response.choices[0].message.content or ""


def run_chat_completion() -> None:
    respan = make_respan(EXAMPLE_NAME)
    client = make_client()
    custom_identifier = make_custom_identifier(EXAMPLE_NAME)
    text = ""

    try:
        with example_attributes(EXAMPLE_NAME, custom_identifier):
            print(f"custom_identifier={custom_identifier}", flush=True)
            print(f"workflow_name={workflow_name(EXAMPLE_NAME)}", flush=True)
            text = _chat_completion_workflow(client)
    finally:
        respan.shutdown()

    print_result(EXAMPLE_NAME, custom_identifier, text)


if __name__ == "__main__":
    run_chat_completion()
